### PR TITLE
Feat/slides/formulario edicion creacion de slides

### PR DESCRIPTION
**Pull request by julian624**
A component who create/edit a slide form, depending if recive or not a slide prop,

i fixed some issues i had with the url of the image in the slide creation
All form inputs are requerided for the post method

### CODE REVIEW

File: src/Components/Slides/SlidesForm.js [hunk 1]
```diff
@@ -1,32 +1,107 @@
...
+    image:slides?.image ?? '',
+    order:slides?.order ?? '',
+  });
+  const hasSlides = !!slides
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+  const handleData = () => {
+    if(!hasSlides){
+    axios({
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: the axios call need to be written on SlidesService that manage the bussiness logic. For example this method here should be createSlide function.

File: src/Components/Slides/SlidesForm.js [hunk 2]
```diff
@@ -1,32 +1,107 @@
...
         }
-    }
-
-    const handleSubmit = (e) =>{
-        e.preventDefault();
-        console.log(initialValues);
-    }
+      })
+      .catch((err) => (err));
+    } else { axios({
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: same of post, this should be on a SlideService and then import the function here. Also if its a post or a put is a business logic so it should be another function in the service,

File: src/Components/Slides/SlidesForm.js [hunk 3]
```diff
@@ -1,32 +1,107 @@
...
+      method: "POST",
+      url: "http://ongapi.alkemy.org/api/slides",
+      data: {
+        name: initialValues.name,
+        description: initialValues.description,
+        image: initialValues.image,
+      },
+    })
+      .then((res) => {
+        if (initialValues.name.length>4) {
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: this validation is a little bit difficult to read. you are validating if the name of the slide length is bigger than 4 after create the slide so it do not make too much sense. This type of validations should be done before save the value.

File: src/Components/Slides/SlidesForm.js [hunk 4]
```diff
@@ -1,32 +1,107 @@
...
+          Swal.fire(
+            'Slide Creado!'
+          )
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: this Swal.fire is frontend logic so on the handleData method you should call something like this:
```
const showSuccessMessage(message = "success") => Swal.fire(message);
const showErrorMessage(message, confirmButtonText) => {
    swal code here
};
...code here

slideService.createOrUpdate(slide)
    .then(()=> showSuccessMessage())
    .catch(error => showErrorMessage(error.errorMessage, 'Ok, voy a editarlo!'));
```

File: src/Components/Slides/SlidesForm.js [hunk 5]
```diff
@@ -1,32 +1,107 @@
...
-        description: ''
-    });
-
-    const handleChange = (e) => {
-        if(e.target.name === 'name'){
-            setInitialValues({...initialValues, name: e.target.value})
-        } if(e.target.name === 'description'){
-            setInitialValues({...initialValues, description: e.target.value})
+const SlidesForm = ({slides}) => {
+  const [initialValues, setInitialValues] = useState({
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: innitial value is too generic name maybe slide is a more specific name

File: src/Components/Slides/SlidesForm.js [hunk 6]
```diff
@@ -1,32 +1,107 @@
...
-        name: '',
-        description: ''
-    });
-
-    const handleChange = (e) => {
-        if(e.target.name === 'name'){
-            setInitialValues({...initialValues, name: e.target.value})
-        } if(e.target.name === 'description'){
-            setInitialValues({...initialValues, description: e.target.value})
+const SlidesForm = ({slides}) => {
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: it receive only one slide or a list of slides? if is only one slide the parameter should be slide, if its a list of slides I believe that slides.name will never work.

File: src/Components/Slides/SlidesForm.js [hunk 7]
```diff
@@ -1,32 +1,107 @@
...
+      ></input>
+      <CKEditor
+        editor={ClassicEditor}
+        className='input-field'
+        type='text'
+        name='description'
+        value={initialValues.description}
+        placeholder='Write some description'
+        required
+        onChange={(e, editor) => {
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js:75]: please, this onChange function should be a separated function

File: src/Components/Slides/SlidesForm.js [hunk 8]
```diff
@@ -1,32 +1,107 @@
...
+        }}
+      />
+      <input
+        className='input-image'
+        type='file'
+        accept='image/png, image/jpg'
+        id='image'
+        name='image'
+        required
+        onChange={(e) => {
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: please this onChange function should be a separated function

File: src/Components/Slides/SlidesForm.js [hunk 9]
```diff
@@ -1,32 +1,107 @@
...
+          const imageFile = e.target.files[0]
+          const imageUrl = new FileReader()
+          imageUrl.readAsDataURL(imageFile)
+          imageUrl.onload = (e) => {
+            setInitialValues({...initialValues, image:e.target.result})
+          }
+        }}
+      />
+      <button onClick={handleData} className="submit-btn" type="submit">
+        {hasSlides ? "Editar" : "Enviar"}
```
Comment (saralavanchy) [inline src/Components/Slides/SlidesForm.js]: good line!!!! nothing to change :)

**Review by saralavanchy — CHANGES_REQUESTED** [review]

File: src/Components/Slides/SlidesForm.js (hunk 1 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js]: changed

File: src/Components/Slides/SlidesForm.js (hunk 2 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js]: Done, i alredy changed it

File: src/Components/Slides/SlidesForm.js (hunk 3 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js]: Done

File: src/Components/Slides/SlidesForm.js (hunk 4 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js]: i was not able to change all  of this, but i tried, maybe you can help me with the succes alert

File: src/Components/Slides/SlidesForm.js (hunk 5 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js]: Already changed to initialSlides

File: src/Components/Slides/SlidesForm.js (hunk 6 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js]: done

File: src/Components/Slides/SlidesForm.js (hunk 7 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js:75]: If i change this to a separated function ckeditor dont work anymore, i dont know how to fix it, maybe making another function for ckeditor in another component? for the moment i didnt changed it

File: src/Components/Slides/SlidesForm.js (hunk 8 above)
Comment (julian624) [reply src/Components/Slides/SlidesForm.js]: Done

**Review by saralavanchy — APPROVED** [review]

**Review by saralavanchy — DISMISSED** [review]

**Review by saralavanchy — APPROVED** [review]

**Pull request merged**